### PR TITLE
RUST-681 Replace lazy static vectors with constant arrays

### PR DESCRIPTION
**Pull request by bugadani**
These arrays aren't modified in run time, so there's no reason for them to be vectors.

### CODE REVIEW

**Comment by agolin95** [comment]
Tracked in https://jira.mongodb.org/browse/RUST-681

File: src/error.rs [hunk 1]
```diff
@@ -555,8 +553,8 @@ pub enum WriteFailure {
     /// An error that occurred due to not being able to satisfy a write concern.
     WriteConcernError(WriteConcernError),
 
-    /// An error that occurred during a write operation that wasn't due to being unable to satisfy a
-    /// write concern.
+    /// An error that occurred during a write operation that wasn't due to being unable to satisfy
+    /// a write concern.
```
Comment (patrickfreed) [inline src/error.rs:557]: This change looks unrelated / unnecessary, could you revert it?

**Review by patrickfreed — COMMENTED** [review]
Thanks for your PR! I just have one small nitpick of a formatting change.

File: src/error.rs (hunk 1 above)
Comment (bugadani) [reply src/error.rs]: Done; automatic code formatting was a bit trigger happy here.

**Review by patrickfreed — APPROVED** [review]
Authorized the CI to run as a sanity check, tagging the other team members in here in the meantime. Thanks again for the PR!

**Review by isabelatkinson — APPROVED** [review]

**Review by saghm — APPROVED** [review]
